src/renderer/vt/VtSequences.cpp

```diff
@@ -464,9 +463,18 @@ using namespace Microsoft::Console::Render;
 // - The hyperlink URI
 // Return Value:
 // - S_OK if we succeeded, else an appropriate HRESULT for failing to allocate or write.
-[[nodiscard]] HRESULT VtEngine::_SetHyperlink(std::wstring uri) noexcept
+[[nodiscard]] HRESULT VtEngine::_SetHyperlink(const std::wstring& uri, const USHORT& id) noexcept
 {
-    const auto convertedUri = ConvertToA(CP_UTF8, uri);
-    const auto hyperlinkFormat = "\x1b]8;;" + convertedUri + "\x7";
-    return _Write(hyperlinkFormat);
+    if (uri.empty())
+    {
+        // Closing OSC8 sequence
+        const auto hyperlinkFormat = (std::wstring)L"\x1b]8;;\x9c";

```

Comment:
```suggestion
        const auto hyperlinkFormat = (std::wstring)L"\x1b]8;;\x9c";
```

const std::wstring hyperlinkFormat{L"\x1b]8;;\x9c"};

We avoid C casts where possible.